packages/testing-lib/src/jest.ts

```diff
@@ -0,0 +1,27 @@
+import * as Context from './context';
+
+type FunctionPropertyNames<T> = {
+  [K in Exclude<keyof T, symbol | number>]: T[K] extends (
+    ...args: Array<any>
+  ) => any
+    ? K
+    : never;
+}[Exclude<keyof T, symbol | number>];
+
+export interface SpyContextManager<
+  T extends {},
+  M extends FunctionPropertyNames<T>
+> {
+  [Context.enter]: () => jest.SpiedFunction<T[M]>;
+  [Context.exit]: (spy: jest.SpiedFunction<T[M]>) => void;
+}
+
+export function spyContext<T extends {}, M extends FunctionPropertyNames<T>>(
```

Comment:
See [packages/testing-lib/src/jest-dom.ts](https://github.com/mongodb/leafygreen-ui/pull/483/files#diff-86a0797df361701293a2a75788e063d6) for usage